### PR TITLE
net: add ng version of the netdev interface

### PR DESCRIPTION
**Pull request by haukepetersen**

### CODE REVIEW

File: sys/include/net/ng_netdev.h [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+     *          NETDEV_MSG_EVENT_TYPE was received
+     *
+     * @param[in] dev           network device descriptor
+     * @param[in] event_type    event type, given by @ref msg_t::value in the
+     *                          received message
+     */
+    void (*isr_event)(ng_netdev_t *dev, uint16_t event_type);
+} ng_netdev_driver_t;
+
+struct ng_netdev_t {
```
Comment (OlegHahm) [inline sys/include/net/ng_netdev.h]: Is there a reason not to typedef this?

Comment (haukepetersen) [inline sys/include/net/ng_netdev.h]: if you look some lines up in the code, you see it is typedefed... The forward declaration is needed as the type is used in `ng_netdev_driver_t`...

Comment (OlegHahm) [inline sys/include/net/ng_netdev.h]: ah, okay, never mind

**Comment by haukepetersen** [comment]
fixed some doxygen

File: sys/include/net/ng_netdev.h [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+#include "net/ng_netconf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Type for @ref msg_t if device fired an event
+ */
+#define NETDEV_MSG_EVENT_TYPE           (0x0100)
```
Comment (cgundogan) [inline sys/include/net/ng_netdev.h]: excuse my ignorance, but where does this number came from?

File: sys/include/net/ng_netdev.h [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+} ng_netdev_driver_t;
+
+/**
+ * @brief   The netdev data-structure holds the minimum information needed for
+ *          interaction with MAC layers and can be expanded with device
+ *          specific fields
+ *
+ * The netdev structure is the parent for all network device driver descriptors.
+ */
+struct ng_netdev_t {
```
Comment (cgundogan) [inline sys/include/net/ng_netdev.h]: I vote for renaming this struct by cutting off the *_t* suffix and adapt line 64. This way the struct definition and the typedef definition don't have the same names.

File: sys/include/net/ng_netdev.h (hunk 2 above)
Comment (haukepetersen) [inline sys/include/net/ng_netdev.h]: nowhere. Just picked some number...

File: sys/include/net/ng_netdev.h (hunk 3 above)
Comment (haukepetersen) [inline sys/include/net/ng_netdev.h]: will do.

**Comment by haukepetersen** [comment]
renamed struct as suggested by @cgundogan

File: sys/include/net/ng_netdev.h (hunk 2 above)
Comment (cgundogan) [inline sys/include/net/ng_netdev.h]: I c.. from the brainware pseudo-random number generator :bowtie:

Comment (haukepetersen) [inline sys/include/net/ng_netdev.h]: yes :-)

Comment (LudwigOrtmann) [inline sys/include/net/ng_netdev.h]: Please surround by `ifndef`

Comment (haukepetersen) [inline sys/include/net/ng_netdev.h]: really? I feel that at some point we start to surround every single define with '#ifndef'... I would rather leave at is and adjust the chosen value if needed -> wasn't there some discussion for system wide message numbers anyway?

**Comment by authmillenon** [comment]
I don't see that the msg type discussion is PR relevant. ACK, please squash.

File: sys/include/net/ng_netdev.h (hunk 2 above)
Comment (LudwigOrtmann) [inline sys/include/net/ng_netdev.h]: Yes, this is the preliminary outcome.

**Comment by haukepetersen** [comment]
squashed and rebased - lets wait for Travis...

**Comment by haukepetersen** [comment]
kicked Travis

**Comment by haukepetersen** [comment]
Travis is happy, and go.

**Pull request merged**